src/cmddroid.cpp

```diff
@@ -80,7 +80,7 @@ void cmdDroidUpdate()
  * It creates a group if it doesn't exist.
  * If the group is not full, it adds the droid to it and sets all the droid's states and orders to the group's.
  */
-void cmdDroidAddDroid(DROID *psCommander, DROID *psDroid)
+void cmdDroidAddDroid(DROID *psCommander, DROID *psDroid, bool playSound /*=true*/)
```

Comment:
Another option would be to return a "result value" from this function (like CMDDROID_ASSIGNED or CMDDROID_EXCEED_MAX_GROUP), and then use this value in order.cpp to play the appropriate sound.